src/lib/stepper/stepper-vertical.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Component, ViewEncapsulation} from '@angular/core';
+import {MdStepper} from './stepper';
+import {animate, state, style, transition, trigger} from '@angular/animations';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-vertical-stepper, mat-vertical-stepper',
+  templateUrl: 'stepper-vertical.html',
+  styleUrls: ['stepper.css'],
+  inputs: ['selectedIndex'],
+  host: {
+    'class': 'mat-stepper-vertical',
+    'role': 'tablist',
+  },
+  animations: [
+      trigger('stepTransition', [
+        state('previous', style({height: '0px', visibility: 'hidden'})),
+        state('next', style({height: '0px', visibility: 'hidden'})),
```

Comment:
Since the `previous` and `next` states have the same styling, you can combine them:

```ts
state('next, previous', style({height: '0px', visibility: 'hidden'}))
```